teacherchannel/database.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import annotations
+
+from typing import List, Optional
+
+from sqlalchemy import BigInteger, Column, ForeignKey, or_, and_, PrimaryKeyConstraint
+from sqlalchemy.orm import relationship
+
+from pie.database import database, session
+
+
+class TeacherChannel(database.base):
+    """Subject channels containing teacher channel, synced from the original channel."""
+
+    __tablename__ = "school_teacherchannel_teacherchannel"
+
+    guild_id = Column(BigInteger)
+    master_id = Column(BigInteger, unique=True)
+    slave_id = Column(BigInteger, primary_key=True, autoincrement=False)
+    teachers = relationship(
+        "Teacher", cascade="all, delete-orphan", back_populates="teacherchannel"
+    )
+
+    @staticmethod
+    def get(guild_id: int, channel_id: int) -> Optional[TeacherChannel]:
+        return (
+            session.query(TeacherChannel)
+            .filter(
+                and_(
+                    or_(
+                        TeacherChannel.master_id == channel_id,
+                        TeacherChannel.slave_id == channel_id,
+                    ),
+                    TeacherChannel.guild_id == guild_id,
+                )
+            )
+            .one_or_none()
+        )
+
+    @staticmethod
+    def get_all(guild_id: int) -> List[TeacherChannel]:
+        return session.query(TeacherChannel).filter_by(guild_id=guild_id).all()
+
+    @staticmethod
+    def get_guild_ids() -> List[TeacherChannel]:
```

Comment:
If I'm not wrong, you are returning list of ID's, not list of TeacherChannel objects, right?